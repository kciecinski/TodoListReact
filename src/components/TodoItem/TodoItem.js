import React from 'react';
import PropTypes from 'prop-types';
import {
  MDBListGroupItem, MDBCol, MDBRow, MDBIcon,
} from 'mdbreact';
import Checkbox from '@material/react-checkbox';

const getItemStyles = (isCompleted) => {
  if (isCompleted) {
    return {
      backgroundColor: '#bbb',
      opacity: '0.5',
      marginTop: '1.5em',
    };
  }
  return {
    marginTop: '1.5em',
  };
};

export default function TodoItem({ todoItem, onRemoveItem, onSetItemCompleted }) {
  const isCompleted = todoItem.completed;

  return (
    <MDBListGroupItem style={{ ...getItemStyles(isCompleted) }}>
      <MDBRow>
        <MDBCol className="todo_items_left" size="10" middle>
          {todoItem.text}
        </MDBCol>
        <MDBCol size="2" className="todo_items_right">
          <Checkbox type="checkbox" onClick={() => onSetItemCompleted(todoItem.id)} disabled={isCompleted} />
          <MDBIcon
            size="2x"
            className="red-text remove_icon"
            label="delete"
            name="delete"
            aria-label="Remove"
            onClick={() => onRemoveItem(todoItem.id)}
            icon="trash"
          />
        </MDBCol>
      </MDBRow>
    </MDBListGroupItem>
  );
}

TodoItem.propTypes = {
  todoItem: PropTypes.shape({
    id: PropTypes.string,
    text: PropTypes.string,
    completed: PropTypes.bool,
  }),
  onRemoveItem: PropTypes.func.isRequired,
  onSetItemCompleted: PropTypes.func.isRequired,
};

TodoItem.defaultProps = {
  todoItem: {},
};
