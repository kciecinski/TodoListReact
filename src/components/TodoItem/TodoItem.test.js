import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TodoItem from './TodoItem';

configure({ adapter: new Adapter() });

describe('<TodoItem />', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      todoItem: {
        completed: false,
        id: 'NJ1HBA5TN',
        text: 'casd',
      },
      removeItem: jest.fn(),
      setTodoDone: jest.fn(),
    };
    wrapper = mount(<TodoItem {...props} />);
  });

  it('renders  correct text', () => {
    expect(wrapper.find('.col-10').text()).toBe('casd');
  });

  it('renders matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders undone', () => {
    expect(wrapper.find('input').props().disabled).toBe(false);
  });
});
