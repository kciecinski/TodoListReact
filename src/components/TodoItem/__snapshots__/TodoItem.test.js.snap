// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> renders matches snapshot 1`] = `
<TodoItem
  onRemoveItem={[MockFunction]}
  onSetItemCompleted={[MockFunction]}
  todoItem={
    Object {
      "completed": false,
      "id": "NJ1HBA5TN",
      "text": "casd",
    }
  }
>
  <ListGroupItem
    style={Object {}}
    tag="li"
  >
    <li
      className="list-group-item"
      style={Object {}}
    >
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            lg={null}
            md={null}
            size="10"
            sm={null}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
            tag="div"
            xl={null}
            xs={null}
          >
            <div
              className="col-10"
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              casd
            </div>
          </Col>
          <Col
            lg={null}
            md={null}
            size="2"
            sm={null}
            style={
              Object {
                "display": "flex",
                "justifyContent": "flex-end",
              }
            }
            tag="div"
            xl={null}
            xs={null}
          >
            <div
              className="col-2"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                }
              }
            >
              <RippledComponent
                checked={false}
                className=""
                disabled={false}
                indeterminate={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                style={Object {}}
                type="checkbox"
                unbounded={true}
              >
                <Checkbox
                  checked={false}
                  className=""
                  disabled={false}
                  indeterminate={false}
                  initRipple={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                  type="checkbox"
                  unbounded={true}
                >
                  <div
                    className="mdc-checkbox mdc-checkbox--upgraded"
                    onAnimationEnd={[Function]}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={Object {}}
                    type="checkbox"
                  >
                    <NativeControl
                      aria-checked="false"
                      checked={false}
                      disabled={false}
                      onChange={[Function]}
                      rippleActivatorRef={
                        Object {
                          "current": <input
                            aria-checked="false"
                            class="mdc-checkbox__native-control"
                            type="checkbox"
                          />,
                        }
                      }
                    >
                      <input
                        aria-checked="false"
                        checked={false}
                        className="mdc-checkbox__native-control"
                        disabled={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                    </NativeControl>
                    <div
                      className="mdc-checkbox__background"
                    >
                      <svg
                        className="mdc-checkbox__checkmark"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          className="mdc-checkbox__checkmark-path"
                          d="M1.73,12.91 8.1,19.28 22.79,4.59"
                          fill="none"
                        />
                      </svg>
                      <div
                        className="mdc-checkbox__mixedmark"
                      />
                    </div>
                  </div>
                </Checkbox>
              </RippledComponent>
              <Fa
                aria-label="Remove"
                border={false}
                brand={false}
                className="red-text"
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="trash"
                inverse=""
                label="delete"
                light={false}
                list={false}
                name="delete"
                onClick={[Function]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  aria-label="Remove"
                  className="fa fa-trash fa-2x red-text"
                  label="delete"
                  name="delete"
                  onClick={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                    }
                  }
                />
              </Fa>
            </div>
          </Col>
        </div>
      </Row>
    </li>
  </ListGroupItem>
</TodoItem>
`;
